diff.go

```diff
@@ -1,57 +1,57 @@
 package gosymbol
 
-func (e constant) D(varName VarName) Expr {
+func (e constant) D(v variable) Expr {
 	return Const(0.0)
 }
 
-func (e variable) D(varName VarName) Expr {
-	if varName == e.Name {
+func (e variable) D(v variable) Expr {
+	if v == e {
```

Comment:
Do you know how this comparison is carried out? I would want it to be true if `e.Name == v.Name`, but it might be good double check that `e == v` isn't just comparing memory locations or something similar.

Reply:
Checked and it works, I can add a test

Reply:
actually this test is already testing that behavior
```	
                 { // Test 2
			input: inputArgs{
				expr:    Var("X"),
				diffVar: Var("X"),
			},
			expectedOutput: Const(1),
		},
```